src/shared/sam/localLambdaRunner.ts

```diff
@@ -328,22 +499,24 @@ export async function attachDebugger({
                     'AWS.output.sam.local.attach.retry.limit.exceeded',
                     'Retry limit reached while trying to attach the debugger.'
                 )
-
-                isDebuggerAttached = false
+                break
             }
         }
-    } while (isDebuggerAttached === undefined)
+    } while (!isDebuggerAttached)
```

Comment:
`isDebuggerAttached` can never be undefined after `onStartDebugging()` assigns it (line 490). So this retry never occurred. 

Fixing this makes tests (and local usage) more reliable. Will revisit later to see if this removes the need for `pythonDebugAdapterHeartbeat.ts` and other timer stuff...